### PR TITLE
Added code action for SemicolonRequiredInReference

### LINKED ISSUE #665: Code action for `SemicolonRequiredInReference`
Given this XML file:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE article [
  <!ENTITY mdash "&#x2014;">
]>
<root>
  &mdash
</root>
```
There is an error for `&mdash` because it misses `;`  we should provide a code action to fix that to insert at the end of  `&mdash`  the missing `;`

### PR DESCRIPTION
**Pull request by JessicaJHee**
lemminx_issue665

Fixes #665 

Signed-off-by: Jessica He <jhe@redhat.com>

### CODE REVIEW

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/codeactions/SemicolonRequiredInReferenceCodeAction.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/**
```
Comment (angelozerr) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/codeactions/SemicolonRequiredInReferenceCodeAction.java:1]: Change header with:

```java
/**
 *  Copyright (c) 2022 Red Hat Inc. and others.
 *  All rights reserved. This program and the accompanying materials
 *  are made available under the terms of the Eclipse Public License v2.0
 *  which accompanies this distribution, and is available at
 *  http://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 *  Contributors:
 *  Red Hat Inc. - initial API and implementation
 */
```

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/codeactions/SemicolonRequiredInReferenceCodeAction.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import org.eclipse.lemminx.services.extensions.ICodeActionParticipant;
+import org.eclipse.lemminx.services.extensions.IComponentProvider;
+import org.eclipse.lemminx.settings.SharedSettings;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.Range;
+
+/**
+ * Code action to fix SemicolonRequiredInReference error.
+ *
```
Comment (angelozerr) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/codeactions/SemicolonRequiredInReferenceCodeAction.java:28]: Add a sample in Javadoc.

**Comment by angelozerr** [comment]
@JessicaJHee please write test in https://github.com/eclipse/lemminx/blob/974099d749010f788f9ac7914894269e38f51a80/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/XMLSyntaxDiagnosticsTest.java#L725 for the 3 tests wich have XMLSyntaxErrorCode.SemicolonRequiredInReference by adding testCodeActionFor (like https://github.com/eclipse/lemminx/blob/974099d749010f788f9ac7914894269e38f51a80/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/XMLSyntaxDiagnosticsTest.java#L764) after each testDiagnostic (ex after

https://github.com/eclipse/lemminx/blob/974099d749010f788f9ac7914894269e38f51a80/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/XMLSyntaxDiagnosticsTest.java#L733)

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/codeactions/SemicolonRequiredInReferenceCodeAction.java (hunk 2 above)
Comment (angelozerr) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/codeactions/SemicolonRequiredInReferenceCodeAction.java:28]: I mean given this XML:

```
<root>
  &mdash
</root>
```

The code action will fix by adding ';':

<root>
  &mdash;
</root>

Comment (angelozerr) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/codeactions/SemicolonRequiredInReferenceCodeAction.java:28]: Indeed the main idea is to see quickly what the code action does. Our current code actions have no this sample and it is hard to understand the behavior of the code action. But let's start with your code action to provide good practice and we will then other code actions with sample.

**Comment by angelozerr** [comment]
Great job @JessicaJHee !

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/codeactions/SemicolonRequiredInReferenceCodeAction.java (hunk 2 above)
Comment (JessicaJHee) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/codeactions/SemicolonRequiredInReferenceCodeAction.java:28]: Sounds good! I can also take some time to update the comments for previous code actions if needed.

**Pull request merged**